roles/pul_nomad/README.md

```diff
@@ -1,36 +1,224 @@
+Princeton's DNS server.
+
+#### Configuration
+
+DNSMasq's main config file is at `/etc/dnsmasq.conf` which we create from [`files/dnsmasq/dnsmasq.conf`](files/dnsmasq/dnsmasq.conf). The main thing we do there is comment out `bind-interfaces` [#5485](https://github.com/pulibrary/princeton_ansible/pull/5485).
+
+There's a second config file at `/etc/dnsmasq.d/dnsmasq-10-consul` which we create from [`templates/dnsmasq/dnsmasq-10-consul.j2`](templates/dnsmasq/dnsmasq-10-consul.j2). It configures Princeton's DNS servers and the fallback to Consul for `.consul` DNS requests.
+
+#### Debugging
+
+To test if DNSMasq is working on a client, run `dig @127.0.0.1 -p 8600 consul.service.consul ANY` and compare it to the results of `dig consul.service.consul ANY`. If they have the same answers, DNSMasq is working.
+
+If you need to flush the DNS Cache because IPs have changed, you can restart DNSMasq with `sudo service dnsmasq restart`
+
+## Nomad
+
+Nomad is a container orchestrator.
+
+What this means is you can write job files, tell Nomad to deploy them, and Nomad
+will automatically find a server in the cluster that can run them. If that
+server goes down it will launch the container on a different server.
+
+We install Podman on our Nomad clients so it can run containers without us
+having to rely on Docker.
+
+### Server Types
+
+Nomad has "servers" and "clients."
+
+We run 3 servers and a larger number of clients.
+
+Servers manage orchestration, receive commands from clients and developers, and
+maintain state about the cluster as a whole. These are named
+`nomad-host-*.princeton.edu`
+
+Clients run jobs. In our case, containers executed by Podman. These are named
+`nomad-client-*.princeton.edu`
+
+### Configuration
+
+Nomad's full configuration reference can be found in their [documentation](https://developer.hashicorp.com/nomad/docs/configuration). You may have to use the table of contents to see details of all the sections.
+
+Every Nomad Client & Server gets `/etc/nomad.d/base.hcl` created from [`templates/nomad/base.hcl.j2`](templates/nomad/base.hcl.j2).
+
+Every Server gets `/etc/nomad.d/server.hcl` created from [`templates/nomad/server.hcl.j2`](templates/nomad/server.hcl.j2).
+
+Every Client gets `/etc/nomad.d/client.hcl` created from [`templates/nomad/client.hcl.j2`](templates/nomad/client.hcl.j2)
+
+The "gossip encryption key" is a key used to encrypt communication between Nomad servers. We use the same key for both Consul & Nomad.
+
+### Consul Interaction
+
+Nomad uses Consul for clustering (the Nomad servers find each other by querying
+Consul) and service discovery (when you launch a job in Nomad it can register a
+service in Consul that we can point to from our load balancers.)
+
+### Host Volumes
+
+Host volumes are a way for containers to save data to the underlying client. It's a folder on the client which the container maps - if a job requires a certain host volume it will only ever get scheduled on a node which has that volume.
```

Comment:
```suggestion
Host volumes are a way for containers to save data to the underlying client VM. It's a folder on the client VM which the container maps - if a job requires a certain host volume it will only ever get scheduled on a node which has that volume.
```